Jenkinsfile.macos

```diff
@@ -1,175 +1,283 @@
+                environment {
+                  PATH = "${env.HOME}/opt/bin:${env.PATH}"
+                }
+                steps {
+                  dir("package/osx/build/src/cpp") {
+                    // attempt to run cpp unit tests
+                    // problems with rsession finding openssl, so those tests
+                    // are disabled until we solve it (#6890)
+                    sh "arch -x86_64 ./rstudio-tests"
+                    
+                    // electron tests not working in CI environment, under investigation
+                    // if(FLAVOR == 'Electron') {
+                    //   sh "cd ../../../../../src/node/desktop/ && $HOME/.yarn/bin/yarn && $HOME/.yarn/bin/yarn test"
+                    // }
+                    }
+                  }
+                  post {
+                    failure {
+                      // mark build as unstable if it fails unit tests
+                      unstable("rstudio-tests failed (${FLAVOR})")
+                    }
+                  }
+                }
+
+              stage('Notarize and Upload') {
+
+                when {
+                  expression {
+                    return params.PUBLISH
+                  }
+                }
+                environment {
+                  PATH = "${env.HOME}/opt/bin:${env.PATH}"
+                  PACKAGE_FILE = ""
+                  BUILD_TYPE = ""
+                  BUILD_DEST = ""
+                  PRODUCT = "${FLAVOR.toLowerCase()}"
+                  RSTUDIO_RELEASE = "${RSTUDIO_VERSION_MAJOR}.${RSTUDIO_VERSION_MINOR}.${RSTUDIO_VERSION_PATCH}${RSTUDIO_VERSION_SUFFIX}"
+                }
+                
+                stages {
+                  stage("Set Release Variables") {
+                    steps {
+                      script {
+                        // extract name of package to publish
+                        PACKAGE_FILE = sh (
+                          script: "basename `ls package/osx/build/RStudio-*.dmg`",
+                          returnStdout: true
+                        ).trim()
+
+                        BUILD_TYPE= sh (
+                          script: "cat version/BUILDTYPE",
+                          returnStdout: true
+                        ).trim().toLowerCase()
+                        if (IS_PRO) {
+                          PRODUCT = PRODUCT + "-pro"
+                        }
+                        BUILD_DEST =  "s3://rstudio-ide-build/${PRODUCT}/macos/"
+                      }
+                    }
+                  }
+
+                  stage("Notarize") {
+                    steps {
+                      withCredentials([usernamePassword(credentialsId: 'ide-apple-notarizer', usernameVariable: 'APPLE_ID', passwordVariable: 'APPLE_ID_PASSWORD')]) {
+                        sh "docker/jenkins/notarize-release.sh package/osx/build/${PACKAGE_FILE}"
+                      }
+                    }
+                  }
+
+                  stage("Upload") {
```

Comment:
We should add a when PUBLISH block here

Reply:
There is a when block for the `PUBLISH` param at the 'Notarize and Upload' parent stage, so all child stages should also get skipped based on that top-level when block.